### DISCUSSION TITLE
Implementing New Function to Merge Prediction Skeletons

### DISCUSSION THREAD

**Discussion by dcmyeates** [talmolab/sleap#1811] [Help!] [answered]
Hello,

I'm currently getting the "ValueError: Labels.skeleton can only be used when there is only a single skeleton saved in the labels" error when trying to combine multiple datasets, as described in a similar post here: https://github.com/talmolab/sleap/discussions/1101. 
While the attached fix outlined by @catubc looks promising, I'm sure how to implement the code in the SLEAP environment currently set up. I'm wondering how the function definitions outlined in the post above are implemented through the command line/script call?

Apologies for the simplicity of the question; this is the first time I've tried to work directly with SLEAP's API like this! Thank you for your help!

```python

def merge_slp_files(fname_slp1, fname_slp2):
    
    #
    fname_hybrid_video = fname_slp1[:-4]+"_merged.avi"
    
    #
    merged_video = sleap.load_video(fname_hybrid_video)
    print(merged_video)

    #
    fname_hybrid_slp = fname_hybrid_video[:-4] + ".slp"

    #
    slp_files = [fname_slp1,
                 fname_slp2
                ]
    
    n_frames = 0
    all_frames = []
    reference_tracks = {}

    first_labels = None
    for slp_file in slp_files:
        
        # Load saved labels.
        labels = sleap.load_file(slp_file)#, match_to=first_labels)
        if first_labels is None:
            first_labels = labels

        new_frames = []
        for i, lf in enumerate(labels):

            # Update reference to merged video.
            lf.video = merged_video

            # Update frame index to the frame number within the merged video.
            lf.frame_idx = n_frames + i

            # # Update the track reference to use the reference tracks to prevent duplication.
            for instance in lf:
                if instance.track is not None:
                    if instance.track.name in reference_tracks:
                        instance.track = reference_tracks[instance.track.name]
                    else:
                        reference_tracks[instance.track.name] = instance.track

            # Append the labeled frame to the list of frames we're keeping from these labels.
            new_frames.append(lf)

        all_frames.extend(new_frames)
        n_frames += len(new_frames)

    merged_labels = sleap.Labels(all_frames)

    # save stuff
    merged_labels.save(fname_hybrid_slp)

    print("DONE...")

# 
merge_slp_files(fname_human_slp,
                fname_idswitch_slp)
```

_Originally posted by @catubc in https://github.com/talmolab/sleap/discussions/1101#discussioncomment-4584526_

**Comment by gitttt-1234** [comment, marked as answer]
Closing this discussion, this is now solved with [`sleap-io` merging functionality](https://io.sleap.ai/latest/merging/)!

Since this was posted, we've built a dedicated merge tool in `sleap-io` that handles skeleton matching, track deduplication, and overlapping frames automatically. The easiest way is from the command line: 

Check `sleap-io` installation [here](https://io.sleap.ai/latest/install/#quick-start)!

```bash
# Basic merge — auto-matches skeletons by node names, deduplicates tracks
sio merge project1.slp predictions.slp -o merged.slp
```

Or via the Python API if you need more control, [here](https://io.sleap.ai/latest/merging/) are some examples!
